src/GovUk.Education.ExploreEducationStatistics.Data.Services/ReleaseService.cs

```diff
@@ -137,14 +128,64 @@ private async Task<List<TableHighlightViewModel>> GetHighlights(Guid releaseId,
             return releaseContentBlocks
                 .Where(dataBlock => subjectsToInclude.Contains(dataBlock.Query.SubjectId))
                 .Select(
-                    dataBlock => new TableHighlightViewModel(
+                    dataBlock => new FeaturedTableViewModel(
                         id: dataBlock.Id,
-                        name: dataBlock.HighlightName,
-                        description: dataBlock.HighlightDescription
+                        name: dataBlock.HighlightName ?? string.Empty,
+                        description: dataBlock.HighlightDescription ?? string.Empty
                     )
                 )
-                .OrderBy(dataBlock => dataBlock.Name)
+                .OrderBy(featuredTable => featuredTable.Name)
+                .ToList();
+        }
+
+        private List<Guid> GetPublishedSubjectIds(Guid releaseId)
+        {
+            return QueryReleaseDataFiles(releaseId)
+                .Join(
+                    _contentDbContext.DataImports,
+                    releaseFile => releaseFile.File,
+                    import => import.File,
+                    (releaseFile, import) => new
+                    {
+                        ReleaseFile = releaseFile,
+                        DataImport = import
+                    }
+                )
+                .Where(join => join.DataImport.Status == DataImportStatus.COMPLETE)
+                .Select(join => join.ReleaseFile.File.SubjectId!.Value)
                 .ToList();
         }
+
+        private IQueryable<ReleaseFile> QueryReleaseDataFiles(Guid releaseId)
+        {
+            return _contentDbContext.ReleaseFiles
+                .Include(rf => rf.File)
+                .Where(
+                    rf => rf.ReleaseId == releaseId
+                          && rf.File.Type == FileType.Data
+                          // Exclude files that are replacements in progress
+                          && !rf.File.ReplacingId.HasValue
+                          && rf.File.SubjectId.HasValue
```

Comment:
It's a shame we have all file types as `File` and can't use inheritance nicely to have a DataFile that has a non-null SubjectId. I think we could do it now but will be limited by EF to use TPH where all the types are still mapped to the same table. SubjectId would still be nullable in the database.

Maybe when we upgrade to EF6 we should refactor File to use TPT table inheritance and change this, removing the need for some of the other stuff around it like unit tests to check it's filtering Files with null Subject Id's.

We could also think about whether two databases and the `Subject` terminology still makes sense or if Observations can be linked direct to DataFile 🙂

Reply:
Yep agreed. Would definitely be nice to get rid of all the implicit rules and checks we currently have to do.

> We could also think about whether two databases and the Subject terminology still makes sense or if Observations can be linked direct to DataFile

Also a good idea! Would be good if we could finally get rid of all the release/subject related stuff, although not sure how feasible that'll end up being :smile: